src/charm.py

```diff
@@ -313,6 +319,55 @@ def _get_interfaces(self):
             raise CheckFailed(err, BlockedStatus)
         return interfaces
 
+    def on_create_profile_action(self, event: ActionEvent) -> None:
+        """Handle the action to create a new profile."""
+        auth_username = event.params.get("auth_username")
```

Comment:
Can we add info log here indicating username, profile, quota?
This will help in debugging in field.